src/js/cf-expandables.js

```diff
@@ -5,110 +5,238 @@
  * A public domain work of the Consumer Financial Protection Bureau
  */
 
-(function( $ ) {
-
-  $.fn.expandable = function( userSettings ) {
-
-    return $( this ).each(function() {
+( function( $ ) {
+  'use strict';
+  var _extend = $.extend;
+
+  /**
+   * Returns function with throttled callback.
+   *
+   * @param {integer} duration - The title of the book.
+   * @param {Function} callback - The title of the book.
+   * @param {object} context - context in which to run callback function.
+   * @returns {Function} - Time delayed function
+   */
+  function _throttle( duration, callback, context ) {
+    var isThrottling = false;
+
+    return function _delay( event ) {
+      if ( isThrottling === true ) {
+        return;
+      }
+      isThrottling = true;
+      callback.call( context, event );
+      window.setTimeout( function() {
+        isThrottling = false;
+      }, duration );
+    };
+  }
+
+  var Expandable = {
+    defaultProperties: {
+      isInAccordion:    false,
+      isExpanded:       false,
+      expandedClass:    'expandable__expanded',
+      throttleDuration: 450
+    },
+
+    // Elements that will be cached.
+    ui: {
+      content: '.expandable_content',
+      target:  '.expandable_target'
+    },
+
+    /**
+     * Creates and customizes the Expandable.
+     *
+     * @param {object} options - Object used to customize the Expandable.
+     */
+    create: function create( options ) {
+      var properties = _extend( true, {}, Expandable.defaultProperties,
+        options );
+
+      // Using $.extend to create an instance of the Expandable.
+      // Extending dom elements is generally bad and should be
+      // avoided but we are doing so to maintain backwards compatibility.
+      // `this` is referring to the dom element.
+      _extend( this, Expandable, properties );
+      this.init();
+    },
+
+    /**
+     * Sets the initial state for the Expandable.
+     */
+    init: function init() {
+      this.isInAccordion = Boolean( this.$el.parents( '.expandable-group' )
```

Comment:
Might break the line at `this.$el.parents` so it's clear `.data()` is chained to that object.